homeassistant/components/solarlog/coordinator.py

```diff
@@ -65,7 +65,8 @@ async def _async_setup(self) -> None:
         _LOGGER.debug("Start async_setup")
         logged_in = False
         if self.solarlog.password != "":
-            logged_in = await self.renew_authentication()
+            if logged_in := await self.renew_authentication():
+                await self.solarlog.test_extended_data_available()
         if logged_in or await self.solarlog.test_extended_data_available():
```

Comment:
Don't we check it twice now?

Reply:
It is my understanding, that on (new) L70 it is only invoked, if logged_in is false, so it is not checked twice.

Reply:
If logged_in is true, it is invoked on L69, otherwise on L70